src/cluster_legacy.c

```diff
@@ -2734,6 +2738,8 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
              * joining another cluster. */
             if (sender && !(flags & CLUSTER_NODE_NOADDR) && !clusterBlacklistExists(g->nodename, CLUSTER_NAMELEN)) {
                 clusterNode *node;
+                // Since shard_ids are learnt only via direct PINGs, flag the node to have an uninitialized shard_id whenever a node entry was added as part of gossip.
```

Comment:
nit. we are still using the c-style comment and the line needs to be less than 120 chars to not fail the clang formatting check
```suggestion
                /* Since shard_ids are learnt only via direct PINGs, flag the node 
                 * to have an uninitialized shard_id whenever a node entry was 
                 * added as part of gossip. */
```